packages/dds/merge-tree/src/sortedSegmentSet.ts

```diff
@@ -55,6 +58,15 @@ export class SortedSegmentSet<
     }
 
     private getOrdinal(item: T): string {
+        const maybeRef = item as Partial<LocalReferencePosition>;
+        if (maybeRef.getSegment !== undefined && maybeRef.isLeaf?.() === false) {
+            const lref = maybeRef as LocalReferencePosition;
+            // Give the local reference a temporary yet valid ordinal.
+            // We pad the ordinal with char(0) to leave space for siblings,
+            // and then add the offset, so the local refs are sorted by
+            // offset which preserves the ordering if the segment splits.
+            return `${lref.getSegment()}${String.fromCharCode(0)}${lref.getOffset()}`;
+        }
```

Comment:
the offset part will give lexicographical sorting of numbers, which isn't what we want here. Best approach I've come up with so far would be to use the fact that segment lengths are bounded and assign some simple mapping from numbers to string suffixes.